flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -178,9 +195,9 @@ public void run() {
 	 * After this method has been called, the coordinator does not accept and further
 	 * messages and cannot trigger any further checkpoints.
 	 */
-	public void shutdown() {
+	public void shutdown() throws Exception {
 		synchronized (lock) {
-			try {	
+			try {
 				if (shutdown) {
 					return;
 				}
```

Comment:
Maybe we can replace this return with simply `if (!shutdown) { the shutdown logic }`